pennylane/operation.py

```diff
@@ -964,6 +964,77 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
```

Comment:
Out of curiosity, what is the motivation to inherit from `Operation` rather than `Operator`? Do we plan on using some of the methods unique to `Operation` such as `generator` or `decomposition`? Do these make sense for a projection?

Reply:
Good question, I'm not familiar with the nuances of `Operation` vs `Operator`, I'm just going by the ADR.